website/docs: install: add aws (#12082)
goauthentik/authentik: website/docs/install-config/install/aws/index.md

---
title: AWS installation
---

> You can install authentik to run on AWS with a Cloudformation template.

Reviewer: *CloudFormation (in all instances below)

### Prerequisites

- An AWS account.
- An [AWS Certificate Manager](https://aws.amazon.com/certificate-manager/) certificate. Take note of the ARN of the certificate.

### Installation

> Create a Cloudformation stack [with our template](https://console.aws.amazon.com/cloudformation/home#/stacks/create/review?stackName=authentik&templateURL=https://authentik-cloudformation-templates.s3.amazonaws.com/authentik.ecs.latest.yaml).

Reviewer: maybe a note that you already need to be logged into AWS to access this template?

Under the **Certificate ARN** input, enter the previously created certificate ARN. You can also configure other settings if needed. You can follow the prompts to create the stack.

> This stack will create the following resources:

Reviewer 2: For a list, we don't need the period at the end of each one.

> - AWS SSM secrets for the PostgreSQL user and the authentik secret key.

Reviewer 2: ```suggestion
- AWS SSM secrets for the PostgreSQL user and the authentik secret key
```

> - A VPC for all other resources.

Reviewer 2: ```suggestion
- A VPC for all other resources
```

> - A RDS PostgreSQL Multi-AZ cluster.

Reviewer 2: ```suggestion
- A RDS PostgreSQL Multi-AZ cluster
```

> - An ElastiCache Redis Multi-AZ cluster.

Reviewer 2: ```suggestion
- An ElastiCache Redis Multi-AZ cluster
```

- An ECS cluster with two tasks:

>     - One for the authentik server.

Reviewer 2: ```suggestion
    - One for the authentik server
```

>     - One for the authentik worker.

Reviewer 2: ```suggestion
    - One for the authentik worker
```

> - An ALB pointing to the authentik server ECS task with the configured certificate.

Reviewer 2: ```suggestion
- An ALB pointing to the authentik server ECS task with the configured certificate
```

> - An EFS filesystem mounted on both ECS tasks for media file storage.

Reviewer 2: ```suggestion
- An EFS filesystem mounted on both ECS tasks for media file storage
```

> The stack will output the endpoint of the ALB that you can point your DNS records to.

Reviewer 2: ```suggestion
The stack will output the endpoint of the ALB (Application Load Balancer) to which you can point your DNS records.
```

> The stack will output the endpoint of the ALB that you can point your DNS records to.

Reviewer 2: ```suggestion
The stack will output the endpoint of the ALB that to which you can point your DNS records.
```
